style(*): grammar, spelling, punctuation (#539)
discordjs/guide: guide/additional-info/changes-in-v12.md

### Image URLs

> Some image-related properties like `user.avatarURL` are now a method in v12 so that you can apply some options to them, e.g., to affect their display size. 

Reviewer: ```suggestion
Some image-related properties like `user.avatarURL` are now methods in v12 so that you can apply some options to them, e.g., to affect their display size. 
```

#### ClientOptions#shards

`clientOptions.shards` was removed and is functionally equivalent to `clientOptions.shardCount` on v12.

#### ClientOptions#sync

`clientOptions.sync` was removed entirely, along with all other user account-only properties and methods.

#### ClientOptions#disabledEvents

`clientOptions.disabledEvents` was removed in favor of using intents. Please refer to our more [detailed article about this topic](/popular-topics/intents)

### ClientUser

#### ClientUser#acceptInvite

`clientUser.acceptInvite()` was removed entirely, along with all other user account-only properties and methods.

#### ClientUser#addFriend

`clientUser.addFriend()` was removed entirely, along with all other user account-only properties and methods.

#### ClientUser#avatarURL

> `clientUser.avatarURL` is now a method, as opposed to a property. It also allows you to determine the file format and size to return. The `dynamic` option will enable you always to get a `.gif` file for animated avatars. Otherwise, the returned link will fall back to the format specified in the `format` option or `.webp` (it's default).

Reviewer: ```suggestion
`clientUser.avatarURL` is now a method, as opposed to a property. It also allows you to determine the file format and size to return. If you provide the `dynamic` option, you will receive a `.gif` URL if the image is animated; otherwise, it will fall back to the specified `format` or its default `.webp`.
```
To match the text from `clientUser#displayAvatarURL`

#### ClientUser#mfaEnabled

`clientUser.mfaEnabled` was removed entirely, along with all other user account-only properties and methods.

#### ClientUser#mobile

`clientUser.mobile` was removed entirely, along with all other user account-only properties and methods.

#### ClientUser#note

`clientUser.note` was removed entirely, along with all other user account-only properties and methods.

#### ClientUser#notes

`clientUser.notes` was removed entirely, along with all other user account-only properties and methods.

#### ClientUser#premium

`clientUser.premium` was removed entirely, along with all other user account-only properties and methods.

#### ClientUser#removeFriend

`clientUser.removeFriend()` was removed entirely, along with all other user account-only properties and methods.

#### ClientUser#send\*\*\*

Just like the `TextChannel#send***` methods, all the `.send***()` methods have been removed in favor of one general `.send()` method. Read through the [TextChannel#send\*\*\*](/additional-info/changes-in-v12.md#channelsend) section for more information.

#### ClientUser#setGame

> `clientUser.setGame()` has been changed to `clientUser.setActivity()`. The second parameter is no longer for providing a streaming URL but rather an object that allows you to provide the URL and activity type.

Reviewer: ```suggestion
`clientUser.setGame()` has changed to `clientUser.setActivity()`. The second parameter is no longer for providing a streaming URL but rather an object that allows you to provide the URL and activity type.
```

#### Guild#acknowledge

`guild.acknowledge()` was removed entirely, along with all other user account-only properties and methods.

#### Guild#allowDMs

`guild.allowDMs()` was removed entirely, along with all other user account-only properties and methods.

#### Guild#ban

> `guild.ban()` has been moved to the `GuildMemberManager`.  In addition, the second parameter in `guild.members.ban()` has been changed. The `options` parameter no longer accepts a number nor a string.

Reviewer: ```suggestion
`guild.ban()` has moved to the `GuildMemberManager`.  In addition, the second parameter in `guild.members.ban()` has changed. The `options` parameter no longer accepts a number nor a string.
```

#### Guild#createChannel

> `guild.createChannel()` has been transformed in the shape of a Manager.  The second, third, and fourth parameters in `guild.createChannel()` have been changed/removed, leaving it with a total of two parameters; the second one is an object with all of the options available in `ChannelData`.

Reviewer: ```suggestion
`guild.createChannel()` has transformed in the shape of a Manager.  The second, third, and fourth parameters in `guild.createChannel()` have changed/removed, leaving it with a total of two parameters; the second one is an object with all of the options available in `ChannelData`.
```

#### Guild#createEmoji

> `guild.createEmoji()` has been transformed in the shape of a Manager.  The third and fourth parameters in `guild.createEmoji()` have been changed/removed, leaving it with a total of three parameters. The `roles` and `reason` parameters from v11 were merged into an object as the third parameter.

Reviewer: ```suggestion
`guild.createEmoji()` has transformed in the shape of a Manager.  The third and fourth parameters in `guild.createEmoji()` were changed/removed, leaving it with a total of three parameters. The `roles` and `reason` parameters from v11 were merged into an object as the third parameter.
```

#### Guild#createRole

> `guild.createRole()` has been transformed in the shape of a Manager.  The first and second parameters in `guild.createRole()` have been changed/removed, leaving it with a total of one parameter. The `data` and `reason` parameters from v11 has moved into an object as the first parameter.

Reviewer: ```suggestion
`guild.createRole()` has transformed in the shape of a Manager.  The first and second parameters in `guild.createRole()` were changed/removed, leaving it with a total of one parameter. The `data` and `reason` parameters from v11 have moved into an object as the first parameter.
```

#### Guild#iconURL

> `guild.iconURL` is now a method, as opposed to a property. It also allows you to determine the file format and size to return. If the `dynamic` option is provided, you will receive a `.gif` URL if the image is animated; otherwise, it will fall back to the specified `format` or its default `.webp` if none is provided.

Reviewer: ```suggestion
`guild.iconURL` is now a method, as opposed to a property. It also allows you to determine the file format and size to return. If you provide the `dynamic` option, you will receive a `.gif` URL if the image is animated; otherwise, it will fall back to the specified `format` or its default `.webp`.
```
To match the other texts.

#### Guild#messageNotifications

`guild.messageNotifications` was removed entirely, along with all other user account-only properties and methods.

#### Guild#mobilePush

`guild.mobilePush` was removed entirely, along with all other user account-only properties and methods.

#### Guild#muted

`guild.muted` was removed entirely, along with all other user account-only properties and methods.

#### Guild#position

`guild.position` was removed entirely, along with all other user account-only properties and methods.

#### Guild#presences

`guild.presences` is now a Manager instead of a Collection.

#### Guild#pruneMembers

> `guild.pruneMembers()` has been transformed in the shape of a Manager.  In addition, the first, second, and third parameters in the method have changed or removed, leaving it with a total of one parameter. The `days`, `dry`, and `reason` parameters from v11 have been merged into an object as the first parameter.

Reviewer: ```suggestion
`guild.pruneMembers()` has transformed in the shape of a Manager.  In addition, the first, second, and third parameters in the method were changed/removed, leaving it with a total of one parameter. The `days`, `dry`, and `reason` parameters from v11 have merged into an object as the first parameter.
```

#### GuildChannel#createInvite

> The second parameter in `channel.createInvite()` was removed, leaving it with a total of one parameter. The `reason` parameter from v11 has been merged into an object as the first parameter.

Reviewer: ```suggestion
The second parameter in `channel.createInvite()` was removed, leaving it with a total of one parameter. The `reason` parameter from v11 has merged into an object as the first parameter.
```

#### GuildChannel#replacePermissionOverwrites

> `guildChannel.replacePermissionOverwrites` has been renamed to `guildChannel.overwritePermissions`. Overwrites and reason are no longer provided Through an options object but directly as method arguments.

Reviewer: ```suggestion
`guildChannel.replacePermissionOverwrites` has been renamed to `guildChannel.overwritePermissions`. Overwrites and reason are no longer provided through an options object; they are now method arguments.
```

#### ShardClientUtil#id

`shardClientUtil.id` was removed and replaced with `shardClientUtil.ids`, which is an array of shard IDs of the current client.

#### ShardClientUtil#singleton

`shardCLientUtil` now has a second parameter, `mode`, to specify whether it's a `process` or `worker`.

### ShardingManager

#### ShardingManger#_spawn

The private method `shardingManager._spawn()` was removed entirely.

#### ShardingManager#createShard

The `id` parameter is now optional and defaults to `this.shards.size`.

#### ShardingManager#launch

The `shardingManager.launch` event was removed entirely and replaced with the `shardingManager.shardCreate` event.

#### ShardingManager#message

The `shardingManager.message` event was removed from this class and is now on the `Shard` class.

#### ShardingManager#respawnAll

> The `waitForReady` parameter has been renamed to `spawnTimeout`, and the `currentShardIndex` parameter was removed entirely.

Reviewer: ```suggestion
The `waitForReady` parameter was renamed to `spawnTimeout`, and the `currentShardIndex` parameter was removed entirely.
```

#### ShardingManager#spawn

> A third, optional parameter, `spawnTimeout`, has been added, specifying how long to wait in milliseconds to wait until the `Client` is ready; the default is `30000`.

Reviewer: ```suggestion
A third, optional parameter, `spawnTimeout`, was added, specifying how long to wait in milliseconds to wait until the `Client` is ready; the default is `30000`.
```

#### TextChannel#acknowledge

> was removed entirely, along with all other user account-only methods and properties.

Reviewer: ```suggestion
`textChannel.acknowledge()`  was removed entirely, along with all other user account-only methods and properties.
```

#### User#addFriend

`user.addFriend()` was removed entirely, along with all other user account-only methods.

#### User#avatarURL

> `user.avatarURL` is now a method, as opposed to a property. It also allows you to determine the file format and size to return. If the `dynamic` option is provided, you will receive a `.gif` URL if the image is animated; otherwise, it will fall back to the specified `format` or its default `.webp` if none is provided.

Reviewer: ```suggestion
`user.avatarURL` is now a method, as opposed to a property. It also allows you to determine the file format and size to return. If you provide the `dynamic` option, you will receive a `.gif` URL if the image is animated; otherwise, it will fall back to the specified `format` or its default `.webp`.
```

#### User#block

`user.block()` was removed entirely, along with all other user account-only methods.

#### User#displayAvatarURL

> `user.displayAvatarURL` is now a method, as opposed to a property. It also allows you to determine the file format and size to return. If the `dynamic` option is provided, you will receive a `.gif` URL if the image is animated; otherwise, it will fall back to the specified `format` or its default `.webp` if none is provided.

Reviewer: ```suggestion
`user.displayAvatarURL` is now a method, as opposed to a property. It also allows you to determine the file format and size to return. If you provide the `dynamic` option, you will receive a `.gif` URL if the image is animated; otherwise, it will fall back to the specified `format` or its default `.webp`.
```

#### VoiceReceiver#destroy

This method was removed entirely; refer to `StreamDispatcher#destroy` for documentation.

#### VoiceReceiver#destroyed

This property was removed entirely.

#### VoiceReceiver#opus

This event was removed entirely.

#### VoiceReceiver#pcm

This event was removed entirely.

#### VoiceReceiver#recreate

This method was removed entirely.

#### VoiceReceiver#voiceConnection

This property was removed entirely.

#### VoiceReceiver#warn

This event was removed entirely; use the `receiver.debug` event instead.

### VoiceRegion

#### VoiceRegion#sampleHostname

This property was removed entirely.

### Webhook

#### Webhook#avatarURL

> `webhook.avatarURL` is now a method, as opposed to a property. It also allows you to determine the file format and size to return. If the `dynamic` option is provided, you will receive a `.gif` URL if the image is animated; otherwise, it will fall back to the specified `format` or its default `.webp` if none is provided.

Reviewer: ```suggestion
`webhook.avatarURL` is now a method, as opposed to a property. It also allows you to determine the file format and size to return. If you provide the `dynamic` option, you will receive a `.gif` URL if the image is animated; otherwise, it will fall back to the specified `format` or its default `.webp`.
```

#### VoiceReceiver#destroy

> This method has been removed entirely; refer to `StreamDispatcher#destroy` for documentation.

Author: The DJS docs don't have docs to `StreamDispatcher#destroy` since it extends [`stream.Writable`](https://nodejs.org/api/stream.html#stream_class_stream_writable), so do you want me to link the docs and rename it to`WritableStream#destroy`, link but don't rename it, or keep it as is?

#### ClientOptions#partials

> `clientOptions.partials` has been added to allow for partial structures–see the `Partials` section of the guide for more details.

Reviewer: ```suggestion
`clientOptions.partials` was added to allow for partial structures–see the `Partials` section of the guide for more details.
```

#### ReactionCollector#remove

The new `remove` event emits when a collected reaction is un-reacted if the `dispose` option is set to `true`.

### Shard

#### Shard#_evals

> The private property `_evals` has been added to map ongoing Promises for calls to `shard.eval()`.

Reviewer: ```suggestion
The private property `_evals` was added to map ongoing Promises for calls to `shard.eval()`.
```

#### Shard#_fetches

> The private property `_fetches` has been added to map ongoing Promises for calls to `shard.fetchClientValues()`.

Reviewer: ```suggestion
The private property `_fetches` was added to map ongoing Promises for calls to `shard.fetchClientValues()`.
```

#### Util#flatten

> This new method flattens any object.  Any `Collection's in the object will be converted to an array of keys.

Reviewer: ```suggestion
This new method flattens any object.  Any `Collection`s in the object will be converted to an array of keys.
```
The old text was correct

### WebSocketShard

> This new class represents a `Shard` 's WebSocket connection.

Reviewer: ```suggestion
This new class represents a `Shard`'s WebSocket connection.
```
